scipy/special/special/cephes/kolmogorov.h

```diff
@@ -0,0 +1,1043 @@
+ *                =  1 -   >         C    d (d + v/n)    (1 - d - v/n)
+ *                         -        n v
+ *                       v=[n(1-d)]+1
+ *
+ * [n(1-d)] is the largest integer not exceeding n(1-d).
+ * nCv is the number of combinations of n things taken v at a time.
+
+ * Sources:
+ * [1] Smirnov, N.V. "Approximate laws of distribution of random variables from empirical data"
+ *     Usp. Mat. Nauk, 1944. http://mi.mathnet.ru/umn8798
+ * [2] Birnbaum, Z. W. and Tingey, Fred H.
+ *     "One-Sided Confidence Contours for Probability Distribution Functions",
+ *     Ann. Math. Statist. 1951.  https://doi.org/10.1214/aoms/1177729550
+ * [3] Dwass, Meyer, "The Distribution of a Generalized $\mathrm{D}^+_n$ Statistic",
+ *     Ann. Math. Statist., 1959.  https://doi.org/10.1214/aoms/1177706085
+ * [4] van Mulbregt, Paul, "Computing the Cumulative Distribution Function and Quantiles of the One-sided
+ Kolmogorov-Smirnov Statistic"
+ *     http://arxiv.org/abs/1802.06966
+ * [5] van Mulbregt, Paul,  "Computing the Cumulative Distribution Function and Quantiles of the limit of the Two-sided
+ Kolmogorov-Smirnov Statistic"
+ *     https://arxiv.org/abs/1803.00426
+ *
+ */
+
+#include "../config.h"
+#include "../error.h"
+
+#include "const.h"
+#include "dd_real.h"
+#include "unity.h"
+
+namespace special {
+namespace cephes {
+
+    namespace detail {
+        /* ************************************************************************ */
+        /* Algorithm Configuration */
+
+        constexpr int KOLMOG_MAXITER = 500;
+
+        /*
+         * Kolmogorov Two-sided:
+         * Switchover between the two series to compute K(x)
+         *  0 <=  x <= KOLMOG_CUTOVER and
+         *  KOLMOG_CUTOVER < x < infty
+         */
+        constexpr double KOLMOG_CUTOVER = 0.82;
+
+        /*
+         * Smirnov One-sided:
+         * n larger than SMIRNOV_MAX_COMPUTE_N will result in an approximation
+         */
+        constexpr int SMIRNOV_MAX_COMPUTE_N = 1000000;
+
+        /*
+         * Use the upper sum formula, if the number of terms is at most SM_UPPER_MAX_TERMS,
+         * and n is at least SM_UPPERSUM_MIN_N
+         * Don't use the upper sum if lots of terms are involved as the series alternates
+         *  sign and the terms get much bigger than 1.
+         */
+        constexpr int SM_UPPER_MAX_TERMS = 3;
+        constexpr int SM_UPPERSUM_MIN_N = 10;
+
+        /* ************************************************************************ */
+        /* ************************************************************************ */
+
+        /* exp() of anything below this returns 0 */
+        constexpr int MIN_EXPABLE = (-708 - 38);
+
+        /* Class to hold the CDF, SF and PDF, which are computed simultaneously */
+        class ThreeProbs {
+          public:
+            double sf;
+            double cdf;
+            double pdf;
+            ThreeProbs(double sf, double cdf, double pdf) : sf(sf), cdf(cdf), pdf(pdf) {}
```

Comment:
With a `struct`, the constructor is not necessary either as the default constructor will just do exactly that using `{...}`.

Reply:
You can just delete the constructor. If there are no constructors, the constructor you have there is implicitly the default when using `{...}`